pkg/k8s/watchers/watcher.go

```diff
@@ -267,9 +267,9 @@ func (k *K8sWatcher) WaitForCRDsToRegister(ctx context.Context) error {
 // caches essential for daemon are synchronized.
 // To be called after WaitForCRDsToRegister() so that all needed CRDs have
 // already been registered.
-func (k *K8sWatcher) InitK8sSubsystem(ctx context.Context) <-chan struct{} {
+func (k *K8sWatcher) InitK8sSubsystem(ctx context.Context, cfg WatcherConfiguration) <-chan struct{} {
```

Comment:
Instead of passing in `cfg` to each function that may use it, we could add it as a field to the `K8sWatcher`